src/include/ipc4/module.h

```diff
@@ -203,6 +206,35 @@ struct ipc4_module_bind_unbind {
 	} extension;
 } __attribute__((packed, aligned(4)));
 
+union ipc4_extended_param_id {
+	uint32_t full;
+	struct{
+		uint32_t parameter_type     : 8;
+		uint32_t parameter_instance : 24;
+	} part;
+};
+
+/* Casting IPC payload to this, address of param_data
+ * becomes the address of the start of actual data.
+ * TLV_DATA_OFFSET is the offset to the actual data from
+ * the beginning of the structure.
+ */
+struct ipc4_mod_conf_param {
+	//! parameter id
+	uint32_t param_id;
+	//! size of value (in bytes)
+	uint32_t param_size;
+	//! value (padded w/ zeroes to align to dword)
+	uint32_t param_data[1];
+};
```

Comment:
We have struct that represent TLV, use it instead.

https://github.com/thesofproject/sof/blob/main/src/include/sof/tlv.h